src/frontend/src/lib/canisters/signer.canister.ts

```diff
@@ -60,12 +62,25 @@ export class SignerCanister extends Canister<SignerService> {
 		return response.Ok.balance;
 	};
 
-	getEthAddress = (): Promise<EthAddress> => {
-		const { caller_eth_address } = this.caller({
+	getEthAddress = async (): Promise<EthAddress> => {
+		const { eth_address_of_caller } = this.caller({
 			certified: true
 		});
 
-		return caller_eth_address();
+		const response = await eth_address_of_caller([
+			{
+				PatronPaysIcrc2Cycles: {
+					owner: Principal.fromText(BACKEND_CANISTER_ID),
```

Comment:
I think this will be quite re-used, no? If yes, shall we put it as constant somewhere to be re-used?

Reply:
Absolutely agree. That's my plan too!

Reply:
I would suggest to wait for next PRs. That way we learn how it is reused before extracting a constant.

Reply:
Every paid API call will  need this so I think this makes sense.  Let me try to find a sensible place for this.